modules/apps/object/object-api/src/main/java/com/liferay/object/system/BaseSystemObjectDefinitionMetadata.java

```diff
@@ -72,6 +73,24 @@ public String getTitleObjectFieldName() {
 		return "id";
 	}
 
+	@Override
+	public long upsertBaseModel(
+			String externalReferenceCode, long companyId, User user,
+			Map<String, Object> values)
+		throws Exception {
+
+		BaseModel<?> baseModel = fetchBaseModelByExternalReferenceCode(
+			externalReferenceCode, companyId);
+
+		if (baseModel != null) {
+			updateBaseModel((Long)baseModel.getPrimaryKeyObj(), user, values);
```

Comment:
Maybe we could do the casting before the method and then, return that casted element:

```
long primaryKey = (long)baseModel.getPrimaryKeyObj();

updateBaseModel(primaryKey, user, values);

return primaryKey;
```